src/core/sock/sock-redirect.cpp

```diff
@@ -866,6 +960,24 @@ extern "C" EXPORT_SYMBOL int listen(int __fd, int backlog)
 {
     srdr_logdbg_entry("fd=%d, backlog=%d", __fd, backlog);
 
+#if defined(DEFINED_ENVOY)
+    if (g_p_app->workers_num > 0) {
+        /* Envoy uses dup() call for listen socket on workers_N (N > 0)
```

Comment:
Could this logic be avoided if we supported dup()? if so, how much effort it requires to support dup?

Reply:
It can be done but current design makes dup() support to hard to do.